packages/core-data/src/components/FacetTimeline.js

```diff
@@ -1,28 +1,502 @@
+          // prevent duplicate ticks
+          .uniq(true, (t) => t.value)
+          .value();
+      }
+      return [];
+    }
+    return majorTicks;
+  }, []);
+
+  /**
+   * On load (or timeline width change), adjust based on the width of the slider.
+   */
+  useEffect(() => {
+    let newMin = range.min;
+    let newMax = range.max;
+
+    // for ticks, expand overall range to get round values
+    // generate major and minor ticks across the range of valid values
+    const majTicks = _.pluck(generateTicks(
+      range.min, range.max, ticksBounds.width, MAJOR_TICKS_MIN_SPACE
+    ), 'value');
+    if (majTicks?.length > 1) {
+      // adjust the min and max such that no values are outside of the
+      // first and last major tick
+      const tickInterval = majTicks[1] - majTicks[0];
+      if (range.min < _.first(majTicks)) {
+        const firstMaj = _.first(majTicks);
+        newMin = Math.min(range.min, firstMaj.getTime() - ONE_DAY - tickInterval);
+      }
+      if (range.max > _.last(majTicks)) {
+        const lastMaj = _.last(majTicks);
+        newMax = Math.max(range.max, lastMaj.getTime() + ONE_DAY + tickInterval);
+      }
+    }
+    setMin(newMin);
+    setMax(newMax);
+  }, [range.min, range.max, ticksBounds.width]);
+
+  /**
+   * Memoize the set of major and minor ticks based on the current min,
+   * max, and slider width.
+   */
+  const ticks = useMemo(() => ({
+    major: ticksBounds?.width ? generateTicks(min, max, ticksBounds.width, 'major') : [],
+    minor: ticksBounds?.width ? generateTicks(min, max, ticksBounds.width, 'minor') : [],
+  }), [min, max, ticksBounds.width]);
+
+  /**
+   * Callback to handle displaying dates on the timeline ticks at different
+   * resolutions (day, month, year).
+   */
+  const displayTickDate = useCallback((date, prevDate) => {
+    if (ticks?.major?.length > 1) {
+      const tickInterval = ticks.major[1].value - ticks.major[0].value;
+      if (!prevDate || prevDate?.getFullYear() !== date.getFullYear()) {
+        // show every new year encountered
+        return date.getFullYear();
+      }
+      if (tickInterval > MIN_MONTH || prevDate || prevDate?.getMonth() !== date.getMonth()) {
+        // less than a year, more than a month between ticks? show month
+        return date.toLocaleDateString(undefined, { month: 'short' });
+      }
+      // less than a month between ticks? show day (numeric)
+      return date.toLocaleDateString(undefined, { day: 'numeric' });
+    }
+    return date.toString();
+  }, [ticks.major]);
+
+  return (
+    <div
+      className={clsx(
+        'grow',
+        'px-4',
+        'pt-6',
+        'pb-2',
+        'flex',
+        'flex-col',
+        'justify-end',
+        'overflow-x-scroll',
+        'select-none',
+        dragging ? 'cursor-grabbing' : 'cursor-grab',
+      )}
+      ref={mergeRefs([frameRef, draggableRef])}
+      onPointerMove={onDrag}
+      onPointerUp={onDragEnd}
+      style={{ scrollbarWidth: 'none' }}
```

Comment:
Can we use a Tailwind class here?

Reply:
Looks like there isn't a Tailwind class for this, but there is a [plugin for it](https://www.npmjs.com/package/tailwind-scrollbar) if we want to go that route. Not sure if it's worth the extra dependency.

Reply:
Nah, this is fine. Thanks for checking into it!